From 956b314cc369d901087fc50074139bb811c92cad Mon Sep 17 00:00:00 2001
From: Chongyun Lee <licy183@termux.dev>
Date: Thu, 17 Apr 2025 03:17:35 +0800
Subject: [PATCH] reland jumbo 5

Enable jumbo build for the following component(s)/source_set(s):

# - //third_party/boringssl:boringssl # Too much to patch
- //third_party/inspector_protocol:crdtp
- //third_party/leveldatabase:leveldatabase
- //net/third_party/quiche:quiche

---
 net/third_party/quiche/BUILD.gn                             | 3 ++-
 net/third_party/quiche/src/quiche/http2/core/spdy_framer.cc | 4 ++++
 .../quiche/src/quiche/http2/core/spdy_protocol.cc           | 4 ++++
 .../src/quiche/quic/core/congestion_control/bbr2_sender.cc  | 4 ++++
 .../src/quiche/quic/core/congestion_control/bbr_sender.cc   | 4 ++++
 .../quic/core/congestion_control/tcp_cubic_sender_bytes.cc  | 4 ++++
 .../src/quiche/quic/core/crypto/aes_128_gcm_12_decrypter.cc | 6 ++++++
 .../src/quiche/quic/core/crypto/aes_128_gcm_12_encrypter.cc | 6 ++++++
 .../src/quiche/quic/core/crypto/aes_128_gcm_decrypter.cc    | 6 ++++++
 .../src/quiche/quic/core/crypto/aes_128_gcm_encrypter.cc    | 6 ++++++
 .../src/quiche/quic/core/crypto/aes_256_gcm_decrypter.cc    | 6 ++++++
 .../src/quiche/quic/core/crypto/aes_256_gcm_encrypter.cc    | 6 ++++++
 .../quiche/quic/core/crypto/chacha20_poly1305_decrypter.cc  | 6 ++++++
 .../quiche/quic/core/crypto/chacha20_poly1305_encrypter.cc  | 6 ++++++
 .../quic/core/crypto/chacha20_poly1305_tls_decrypter.cc     | 6 ++++++
 .../quic/core/crypto/chacha20_poly1305_tls_encrypter.cc     | 6 ++++++
 .../src/quiche/quic/core/crypto/crypto_secret_boxer.cc      | 4 ++++
 .../quiche/src/quiche/quic/core/http/web_transport_http3.cc | 2 ++
 net/third_party/quiche/src/quiche/quic/core/quic_stream.cc  | 2 ++
 .../quiche/src/quiche/quic/core/quic_stream_id_manager.cc   | 2 ++
 .../quiche/src/quiche/quic/core/tls_handshaker.cc           | 2 ++
 third_party/inspector_protocol/BUILD.gn                     | 4 +++-
 third_party/leveldatabase/BUILD.gn                          | 3 ++-
 23 files changed, 99 insertions(+), 3 deletions(-)

diff --git a/net/third_party/quiche/BUILD.gn b/net/third_party/quiche/BUILD.gn
index a48a70817a..731253b8f0 100644
--- a/net/third_party/quiche/BUILD.gn
+++ b/net/third_party/quiche/BUILD.gn
@@ -24,6 +24,7 @@
 #
 #             (*) only used on platforms where epoll tests are built
 
+import("//build/config/jumbo.gni")
 import("//net/third_party/quiche/src/build/source_list.gni")
 import("//testing/libfuzzer/fuzzer_test.gni")
 import("//third_party/protobuf/proto_library.gni")
@@ -50,7 +51,7 @@ config("quiche_config") {
   ]
 }
 
-component("quiche") {
+jumbo_component("quiche") {
   sources = [
               "overrides/quiche_platform_impl/quiche_bug_tracker_impl.h",
               "overrides/quiche_platform_impl/quiche_client_stats_impl.h",
diff --git a/net/third_party/quiche/src/quiche/http2/core/spdy_framer.cc b/net/third_party/quiche/src/quiche/http2/core/spdy_framer.cc
index 9f400c43e6..faf49e76a2 100644
--- a/net/third_party/quiche/src/quiche/http2/core/spdy_framer.cc
+++ b/net/third_party/quiche/src/quiche/http2/core/spdy_framer.cc
@@ -41,6 +41,8 @@ uint32_t PackStreamDependencyValues(bool exclusive,
 // Used to indicate no flags in a HTTP2 flags field.
 const uint8_t kNoFlags = 0;
 
+#define kPadLengthFieldSize kPadLengthFieldSize_SpdyFramer
+
 // Wire size of pad length field.
 const size_t kPadLengthFieldSize = 1;
 
@@ -1376,3 +1378,5 @@ size_t SpdyFramer::header_encoder_table_size() const {
 }
 
 }  // namespace spdy
+
+#undef kPadLengthFieldSize
diff --git a/net/third_party/quiche/src/quiche/http2/core/spdy_protocol.cc b/net/third_party/quiche/src/quiche/http2/core/spdy_protocol.cc
index 018b8c9c55..d7ad72591d 100644
--- a/net/third_party/quiche/src/quiche/http2/core/spdy_protocol.cc
+++ b/net/third_party/quiche/src/quiche/http2/core/spdy_protocol.cc
@@ -20,6 +20,8 @@
 #include "quiche/common/platform/api/quiche_flag_utils.h"
 #include "quiche/common/platform/api/quiche_logging.h"
 
+#define kPadLengthFieldSize kPadLengthFieldSize_SpdyProtocol
+
 namespace spdy {
 
 const char* const kHttp2ConnectionHeaderPrefix =
@@ -612,3 +614,5 @@ size_t GetPushPromiseFrameSizeSansBlock(
 }
 
 }  // namespace spdy
+
+#undef kPadLengthFieldSize
diff --git a/net/third_party/quiche/src/quiche/quic/core/congestion_control/bbr2_sender.cc b/net/third_party/quiche/src/quiche/quic/core/congestion_control/bbr2_sender.cc
index 21134ac65e..1d357e6e4b 100644
--- a/net/third_party/quiche/src/quiche/quic/core/congestion_control/bbr2_sender.cc
+++ b/net/third_party/quiche/src/quiche/quic/core/congestion_control/bbr2_sender.cc
@@ -23,6 +23,8 @@
 #include "quiche/common/platform/api/quiche_logging.h"
 #include "quiche/common/print_elements.h"
 
+#define kDefaultMinimumCongestionWindow kDefaultMinimumCongestionWindow_BBR2
+
 namespace quic {
 
 namespace {
@@ -584,3 +586,5 @@ std::ostream& operator<<(std::ostream& os, const Bbr2Sender::DebugState& s) {
 }
 
 }  // namespace quic
+
+#undef kDefaultMinimumCongestionWindow
diff --git a/net/third_party/quiche/src/quiche/quic/core/congestion_control/bbr_sender.cc b/net/third_party/quiche/src/quiche/quic/core/congestion_control/bbr_sender.cc
index 34bb0689a3..01ca5ab92f 100644
--- a/net/third_party/quiche/src/quiche/quic/core/congestion_control/bbr_sender.cc
+++ b/net/third_party/quiche/src/quiche/quic/core/congestion_control/bbr_sender.cc
@@ -19,6 +19,8 @@
 #include "quiche/quic/platform/api/quic_flags.h"
 #include "quiche/quic/platform/api/quic_logging.h"
 
+#define kDefaultMinimumCongestionWindow kDefaultMinimumCongestionWindow_BBR
+
 namespace quic {
 
 namespace {
@@ -895,3 +897,5 @@ std::ostream& operator<<(std::ostream& os, const BbrSender::DebugState& state) {
 }
 
 }  // namespace quic
+
+#undef kDefaultMinimumCongestionWindow
diff --git a/net/third_party/quiche/src/quiche/quic/core/congestion_control/tcp_cubic_sender_bytes.cc b/net/third_party/quiche/src/quiche/quic/core/congestion_control/tcp_cubic_sender_bytes.cc
index 610bce71cb..6a0ad1c616 100644
--- a/net/third_party/quiche/src/quiche/quic/core/congestion_control/tcp_cubic_sender_bytes.cc
+++ b/net/third_party/quiche/src/quiche/quic/core/congestion_control/tcp_cubic_sender_bytes.cc
@@ -16,6 +16,8 @@
 #include "quiche/quic/platform/api/quic_flags.h"
 #include "quiche/quic/platform/api/quic_logging.h"
 
+#define kDefaultMinimumCongestionWindow kDefaultMinimumCongestionWindow_CUBIC
+
 namespace quic {
 
 namespace {
@@ -385,3 +387,5 @@ CongestionControlType TcpCubicSenderBytes::GetCongestionControlType() const {
 }
 
 }  // namespace quic
+
+#undef kDefaultMinimumCongestionWindow
diff --git a/net/third_party/quiche/src/quiche/quic/core/crypto/aes_128_gcm_12_decrypter.cc b/net/third_party/quiche/src/quiche/quic/core/crypto/aes_128_gcm_12_decrypter.cc
index 7842508b1e..4b83776968 100644
--- a/net/third_party/quiche/src/quiche/quic/core/crypto/aes_128_gcm_12_decrypter.cc
+++ b/net/third_party/quiche/src/quiche/quic/core/crypto/aes_128_gcm_12_decrypter.cc
@@ -7,6 +7,9 @@
 #include "openssl/aead.h"
 #include "openssl/tls1.h"
 
+#define kKeySize   kKeySize_AES128GCM12Dec
+#define kNonceSize kNonceSize_AES128GCM12Dec
+
 namespace quic {
 
 namespace {
@@ -31,3 +34,6 @@ uint32_t Aes128Gcm12Decrypter::cipher_id() const {
 }
 
 }  // namespace quic
+
+#undef kKeySize
+#undef kNonceSize
diff --git a/net/third_party/quiche/src/quiche/quic/core/crypto/aes_128_gcm_12_encrypter.cc b/net/third_party/quiche/src/quiche/quic/core/crypto/aes_128_gcm_12_encrypter.cc
index 36275a24d1..32a032e347 100644
--- a/net/third_party/quiche/src/quiche/quic/core/crypto/aes_128_gcm_12_encrypter.cc
+++ b/net/third_party/quiche/src/quiche/quic/core/crypto/aes_128_gcm_12_encrypter.cc
@@ -6,6 +6,9 @@
 
 #include "openssl/evp.h"
 
+#define kKeySize   kKeySize_AES128GCM12Enc
+#define kNonceSize kNonceSize_AES128GCM12Enc
+
 namespace quic {
 
 namespace {
@@ -26,3 +29,6 @@ Aes128Gcm12Encrypter::Aes128Gcm12Encrypter()
 Aes128Gcm12Encrypter::~Aes128Gcm12Encrypter() {}
 
 }  // namespace quic
+
+#undef kKeySize
+#undef kNonceSize
diff --git a/net/third_party/quiche/src/quiche/quic/core/crypto/aes_128_gcm_decrypter.cc b/net/third_party/quiche/src/quiche/quic/core/crypto/aes_128_gcm_decrypter.cc
index de3e6dec28..11bb0389af 100644
--- a/net/third_party/quiche/src/quiche/quic/core/crypto/aes_128_gcm_decrypter.cc
+++ b/net/third_party/quiche/src/quiche/quic/core/crypto/aes_128_gcm_decrypter.cc
@@ -9,6 +9,9 @@
 #include "quiche/quic/platform/api/quic_flag_utils.h"
 #include "quiche/quic/platform/api/quic_flags.h"
 
+#define kKeySize   kKeySize_AES128GCMDec
+#define kNonceSize kNonceSize_AES128GCMDec
+
 namespace quic {
 
 namespace {
@@ -33,3 +36,6 @@ uint32_t Aes128GcmDecrypter::cipher_id() const {
 }
 
 }  // namespace quic
+
+#undef kKeySize
+#undef kNonceSize
diff --git a/net/third_party/quiche/src/quiche/quic/core/crypto/aes_128_gcm_encrypter.cc b/net/third_party/quiche/src/quiche/quic/core/crypto/aes_128_gcm_encrypter.cc
index 101888fefa..4986886028 100644
--- a/net/third_party/quiche/src/quiche/quic/core/crypto/aes_128_gcm_encrypter.cc
+++ b/net/third_party/quiche/src/quiche/quic/core/crypto/aes_128_gcm_encrypter.cc
@@ -6,6 +6,9 @@
 
 #include "openssl/evp.h"
 
+#define kKeySize   kKeySize_AES128GCMEnc
+#define kNonceSize kNonceSize_AES128GCMEnc
+
 namespace quic {
 
 namespace {
@@ -26,3 +29,6 @@ Aes128GcmEncrypter::Aes128GcmEncrypter()
 Aes128GcmEncrypter::~Aes128GcmEncrypter() {}
 
 }  // namespace quic
+
+#undef kKeySize
+#undef kNonceSize
diff --git a/net/third_party/quiche/src/quiche/quic/core/crypto/aes_256_gcm_decrypter.cc b/net/third_party/quiche/src/quiche/quic/core/crypto/aes_256_gcm_decrypter.cc
index 8aa15f1fca..ff4408366d 100644
--- a/net/third_party/quiche/src/quiche/quic/core/crypto/aes_256_gcm_decrypter.cc
+++ b/net/third_party/quiche/src/quiche/quic/core/crypto/aes_256_gcm_decrypter.cc
@@ -9,6 +9,9 @@
 #include "quiche/quic/platform/api/quic_flag_utils.h"
 #include "quiche/quic/platform/api/quic_flags.h"
 
+#define kKeySize   kKeySize_AES256GCMDec
+#define kNonceSize kNonceSize_AES256GCMDec
+
 namespace quic {
 
 namespace {
@@ -33,3 +36,6 @@ uint32_t Aes256GcmDecrypter::cipher_id() const {
 }
 
 }  // namespace quic
+
+#undef kKeySize
+#undef kNonceSize
diff --git a/net/third_party/quiche/src/quiche/quic/core/crypto/aes_256_gcm_encrypter.cc b/net/third_party/quiche/src/quiche/quic/core/crypto/aes_256_gcm_encrypter.cc
index 27cb3831be..449847172e 100644
--- a/net/third_party/quiche/src/quiche/quic/core/crypto/aes_256_gcm_encrypter.cc
+++ b/net/third_party/quiche/src/quiche/quic/core/crypto/aes_256_gcm_encrypter.cc
@@ -6,6 +6,9 @@
 
 #include "openssl/evp.h"
 
+#define kKeySize   kKeySize_AES256GCMEnc
+#define kNonceSize kNonceSize_AES256GCMEnc
+
 namespace quic {
 
 namespace {
@@ -26,3 +29,6 @@ Aes256GcmEncrypter::Aes256GcmEncrypter()
 Aes256GcmEncrypter::~Aes256GcmEncrypter() {}
 
 }  // namespace quic
+
+#undef kKeySize
+#undef kNonceSize
diff --git a/net/third_party/quiche/src/quiche/quic/core/crypto/chacha20_poly1305_decrypter.cc b/net/third_party/quiche/src/quiche/quic/core/crypto/chacha20_poly1305_decrypter.cc
index 9ec7f86eca..57610af09b 100644
--- a/net/third_party/quiche/src/quiche/quic/core/crypto/chacha20_poly1305_decrypter.cc
+++ b/net/third_party/quiche/src/quiche/quic/core/crypto/chacha20_poly1305_decrypter.cc
@@ -7,6 +7,9 @@
 #include "openssl/aead.h"
 #include "openssl/tls1.h"
 
+#define kKeySize   kKeySize_ChaCha20Poly1305Dec
+#define kNonceSize kNonceSize_ChaCha20Poly1305Dec
+
 namespace quic {
 
 namespace {
@@ -39,3 +42,6 @@ QuicPacketCount ChaCha20Poly1305Decrypter::GetIntegrityLimit() const {
 }
 
 }  // namespace quic
+
+#undef kKeySize
+#undef kNonceSize
diff --git a/net/third_party/quiche/src/quiche/quic/core/crypto/chacha20_poly1305_encrypter.cc b/net/third_party/quiche/src/quiche/quic/core/crypto/chacha20_poly1305_encrypter.cc
index fca4b85d4a..6f962fcbfd 100644
--- a/net/third_party/quiche/src/quiche/quic/core/crypto/chacha20_poly1305_encrypter.cc
+++ b/net/third_party/quiche/src/quiche/quic/core/crypto/chacha20_poly1305_encrypter.cc
@@ -8,6 +8,9 @@
 
 #include "openssl/evp.h"
 
+#define kKeySize   kKeySize_ChaCha20Poly1305Enc
+#define kNonceSize kNonceSize_ChaCha20Poly1305Enc
+
 namespace quic {
 
 namespace {
@@ -35,3 +38,6 @@ QuicPacketCount ChaCha20Poly1305Encrypter::GetConfidentialityLimit() const {
 }
 
 }  // namespace quic
+
+#undef kKeySize
+#undef kNonceSize
diff --git a/net/third_party/quiche/src/quiche/quic/core/crypto/chacha20_poly1305_tls_decrypter.cc b/net/third_party/quiche/src/quiche/quic/core/crypto/chacha20_poly1305_tls_decrypter.cc
index 623304d7e2..b0308c573d 100644
--- a/net/third_party/quiche/src/quiche/quic/core/crypto/chacha20_poly1305_tls_decrypter.cc
+++ b/net/third_party/quiche/src/quiche/quic/core/crypto/chacha20_poly1305_tls_decrypter.cc
@@ -9,6 +9,9 @@
 #include "quiche/quic/platform/api/quic_flag_utils.h"
 #include "quiche/quic/platform/api/quic_flags.h"
 
+#define kKeySize   kKeySize_ChaCha20Poly1305TLSDec
+#define kNonceSize kNonceSize_ChaCha20Poly1305TLSDec
+
 namespace quic {
 
 namespace {
@@ -41,3 +44,6 @@ QuicPacketCount ChaCha20Poly1305TlsDecrypter::GetIntegrityLimit() const {
 }
 
 }  // namespace quic
+
+#undef kKeySize
+#undef kNonceSize
diff --git a/net/third_party/quiche/src/quiche/quic/core/crypto/chacha20_poly1305_tls_encrypter.cc b/net/third_party/quiche/src/quiche/quic/core/crypto/chacha20_poly1305_tls_encrypter.cc
index e47242b998..59c942ff44 100644
--- a/net/third_party/quiche/src/quiche/quic/core/crypto/chacha20_poly1305_tls_encrypter.cc
+++ b/net/third_party/quiche/src/quiche/quic/core/crypto/chacha20_poly1305_tls_encrypter.cc
@@ -8,6 +8,9 @@
 
 #include "openssl/evp.h"
 
+#define kKeySize   kKeySize_ChaCha20Poly1305TLSEnc
+#define kNonceSize kNonceSize_ChaCha20Poly1305TLSEnc
+
 namespace quic {
 
 namespace {
@@ -35,3 +38,6 @@ QuicPacketCount ChaCha20Poly1305TlsEncrypter::GetConfidentialityLimit() const {
 }
 
 }  // namespace quic
+
+#undef kKeySize
+#undef kNonceSize
diff --git a/net/third_party/quiche/src/quiche/quic/core/crypto/crypto_secret_boxer.cc b/net/third_party/quiche/src/quiche/quic/core/crypto/crypto_secret_boxer.cc
index bdc87e0594..af6cb33b0b 100644
--- a/net/third_party/quiche/src/quiche/quic/core/crypto/crypto_secret_boxer.cc
+++ b/net/third_party/quiche/src/quiche/quic/core/crypto/crypto_secret_boxer.cc
@@ -17,6 +17,8 @@
 #include "quiche/quic/core/crypto/quic_random.h"
 #include "quiche/quic/platform/api/quic_logging.h"
 
+#define kAEAD kAEAD_CryptoSecretBoxer
+
 namespace quic {
 
 // kSIVNonceSize contains the number of bytes of nonce in each AES-GCM-SIV box.
@@ -146,3 +148,5 @@ bool CryptoSecretBoxer::Unbox(absl::string_view in_ciphertext,
 }
 
 }  // namespace quic
+
+#undef kAEAD
diff --git a/net/third_party/quiche/src/quiche/quic/core/http/web_transport_http3.cc b/net/third_party/quiche/src/quiche/quic/core/http/web_transport_http3.cc
index 86d60860f9..24d911aa87 100644
--- a/net/third_party/quiche/src/quiche/quic/core/http/web_transport_http3.cc
+++ b/net/third_party/quiche/src/quiche/quic/core/http/web_transport_http3.cc
@@ -514,3 +514,5 @@ WebTransportHttp3::MaybeSetSubprotocolFromResponseHeaders(
 }
 
 }  // namespace quic
+
+#undef ENDPOINT
diff --git a/net/third_party/quiche/src/quiche/quic/core/quic_stream.cc b/net/third_party/quiche/src/quiche/quic/core/quic_stream.cc
index 92df3832f8..d9f92fc4f4 100644
--- a/net/third_party/quiche/src/quiche/quic/core/quic_stream.cc
+++ b/net/third_party/quiche/src/quiche/quic/core/quic_stream.cc
@@ -1646,3 +1646,5 @@ QuicByteCount QuicStream::CalculateSendWindowSize() const {
 }
 
 }  // namespace quic
+
+#undef ENDPOINT
diff --git a/net/third_party/quiche/src/quiche/quic/core/quic_stream_id_manager.cc b/net/third_party/quiche/src/quiche/quic/core/quic_stream_id_manager.cc
index 0129beaec4..e3f0d88d57 100644
--- a/net/third_party/quiche/src/quiche/quic/core/quic_stream_id_manager.cc
+++ b/net/third_party/quiche/src/quiche/quic/core/quic_stream_id_manager.cc
@@ -244,3 +244,5 @@ QuicStreamCount QuicStreamIdManager::available_incoming_streams() const {
 }
 
 }  // namespace quic
+
+#undef ENDPOINT
diff --git a/net/third_party/quiche/src/quiche/quic/core/tls_handshaker.cc b/net/third_party/quiche/src/quiche/quic/core/tls_handshaker.cc
index 9529230d94..b80e76f7e9 100644
--- a/net/third_party/quiche/src/quiche/quic/core/tls_handshaker.cc
+++ b/net/third_party/quiche/src/quiche/quic/core/tls_handshaker.cc
@@ -474,3 +474,5 @@ void TlsHandshaker::MessageCallback(bool is_write, int /*version*/,
 }
 
 }  // namespace quic
+
+#undef ENDPOINT
diff --git a/third_party/inspector_protocol/BUILD.gn b/third_party/inspector_protocol/BUILD.gn
index 551df456f5..3b3039fbdb 100644
--- a/third_party/inspector_protocol/BUILD.gn
+++ b/third_party/inspector_protocol/BUILD.gn
@@ -2,7 +2,9 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
-component("crdtp") {
+import("//build/config/jumbo.gni")
+
+jumbo_component("crdtp") {
   sources = [
     "crdtp/cbor.cc",
     "crdtp/cbor.h",
diff --git a/third_party/leveldatabase/BUILD.gn b/third_party/leveldatabase/BUILD.gn
index db16e21024..dbf02f0dee 100644
--- a/third_party/leveldatabase/BUILD.gn
+++ b/third_party/leveldatabase/BUILD.gn
@@ -2,6 +2,7 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
+import("//build/config/jumbo.gni")
 import("//testing/libfuzzer/fuzzer_test.gni")
 import("//testing/test.gni")
 
@@ -112,7 +113,7 @@ config("leveldb_component_config") {
   }
 }
 
-component("leveldatabase") {
+jumbo_component("leveldatabase") {
   sources = leveldb_sources
 
   configs -= [ "//build/config/compiler:chromium_code" ]
